queue/scope.go

```diff
@@ -48,6 +49,10 @@ func (s *scopeManagerImpl) Acquire(id string, scopes []string) error {
 		return errors.Errorf("could not acquire lock scope '%s' held by '%s' not '%s'", sc, holder, id)
 	}
 
+	for _, sc := range scopesToAcquire {
```

Comment:
I am confused why you moved this down here instead of keeping it in the for loop above?

Reply:
If you don't do this, you can end up partially acquiring all of the scopes you asked for. If you ask for the "foo" and "bar" scopes but "bar" is already in use by a different job, the old code would successfully acquire "foo" and error trying to acquire "bar". This change is to make acquiring all of the scopes atomic (and i wrote a test for it) - either you get all the scopes you ask for, or none of them.